molgenis-data/src/main/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorFactory.java

```diff
@@ -0,0 +1,24 @@
+package org.molgenis.data.decorator;
+
+import org.molgenis.data.Entity;
+import org.molgenis.data.Repository;
+import org.molgenis.data.meta.model.EntityType;
+
+/**
+ * Repository decorator factory that creates decorated {@link Repository repositories} for specific entity types.
+ * Used to decorate (system) entity types dynamically.
```

Comment:
Would it make sense to merge system repository decorators with dynamic decorators?

Reply:
I've thought about it but that would mean that it would be possible to remove system critical decorators from system entities, among other things. The way I see it, 'custom' decorators are specific to one SystemEntityType and 'dynamic' decorators should be generally useful for more than one EntityType.

Reply:
> but that would mean that it would be possible to remove system critical decorators from system entities

Can be prevented using a check similar to EntityTypeRepositorySecurityDecorator.validateDeleteAllowed

Reply:
I'm not sure of the benefit of introducing extra logic whilst exposing the inner workings of system decorators to admins/users who won't be able to change or use them anyways. System decorators really are different from dynamic decorators if you ask me and it's best to keep them separated.